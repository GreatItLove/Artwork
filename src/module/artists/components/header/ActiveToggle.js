import React from 'react';
import { connect } from 'react-redux';
import Toggle from 'material-ui/Toggle';
import { compose } from 'recompose';
import muiThemeable from 'material-ui/styles/muiThemeable';
import PropTypes from 'prop-types';

import { patchArtistRequest } from '../../artistsActions';
const styles = {
  toggle: {
    marginBottom: 16,
  },
  thumbSwitched: {
    backgroundColor: '#E18928',
  },
  trackSwitched: {
    backgroundColor: '#FEBE69',
  },
  labelStyle: {
    color: '#8F9082'
  }
};

const propTypes = {
  muiTheme: PropTypes.shape({
    palette: PropTypes.shape({
      colorBrandOrange: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  activeArtist: PropTypes.func.isRequired,
};

/**
 * @function ActiveToggle
 * @param {Object} muiTheme material ui theme object
 * @param {Function} activeArtist
 * @returns {XML}
 */
function ActiveToggle({ muiTheme, activeArtist, active, artistsId }) {
  return (
    <Toggle
      label="ACTIVE"
      labelStyle={styles.labelStyle}
      style={styles.toggle}
      thumbSwitchedStyle={styles.thumbSwitched}
      trackSwitchedStyle={styles.trackSwitched}
      defaultToggled={active}
      onToggle={(obj, b) => { activeArtist({ active: b, artistsId: artistsId }); }}
    />
  );
}

ActiveToggle.propTypes = propTypes;

const mapDispatchToProps = dispatch => ({
  activeArtist: data => {
    const { artistsId, ...rest } = data;
    dispatch(patchArtistRequest(rest, artistsId));
  },
});

const mapStateToProps = state => ({
  active: state.artists.artistInfo.active,
  artistsId: state.artists.artistInfo.artistsId
})

const enhance = compose(muiThemeable(), connect(mapStateToProps, mapDispatchToProps));

export { ActiveToggle as ActiveToggleTest };

export default enhance(ActiveToggle);
