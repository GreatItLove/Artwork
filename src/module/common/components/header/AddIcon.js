import React from 'react';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import FontIcon from 'material-ui/FontIcon';
import muiThemeable from 'material-ui/styles/muiThemeable';
import PropTypes from 'prop-types';

const propTypes = {
  muiTheme: PropTypes.shape({
    palette: PropTypes.shape({
      colorBrandOrange: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  openAddDialog: PropTypes.func
};

/**
 * @function AddIcon
 * @param {Object} muiTheme material ui theme object
 * @param {Function} openAddDialog
 * @returns {XML}
 */
function AddIcon({ muiTheme, openAddDialog }) {
  return (
    <FloatingActionButton
      mini
      backgroundColor={muiTheme.palette.colorBrandOrange}
      style={{ boxShadow: 'none' }}
      children={<FontIcon className="material-icons">add</FontIcon>}
      onClick={openAddDialog}
    />
  );
}

AddIcon.propTypes = propTypes;

export { AddIcon as AddIconTest };

export default muiThemeable()(AddIcon);
