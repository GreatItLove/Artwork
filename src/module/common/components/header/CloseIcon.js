import React from 'react';
import FontIcon from 'material-ui/FontIcon';
import PropTypes from 'prop-types';

const propTypes = {
  closeDialog: PropTypes.func.isRequired,
};

function CloseIcon({ closeDialog }) {
  return (
    <FontIcon
      className="material-icons"
      onClick={closeDialog}
      style={{ 'fontSize': 36, 'color': '#66a4c0', 'cursor': 'pointer' }}>
      close
    </FontIcon>
  );
}

CloseIcon.propTypes = propTypes;

export default CloseIcon;
