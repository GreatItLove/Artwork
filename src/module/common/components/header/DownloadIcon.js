import React from 'react';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import FontIcon from 'material-ui/FontIcon';
import muiThemeable from 'material-ui/styles/muiThemeable';
import PropTypes from 'prop-types';

const propTypes = {
  muiTheme: PropTypes.shape({
    palette: PropTypes.shape({
      colorBrandOrange: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  openDownloadDialog: PropTypes.func,
};

/**
 * @function DownloadIcon
 * @returns {XML}
 */
function DownloadIcon({ muiTheme, openDownloadDialog }) {
  return (
    <FloatingActionButton
      mini
      backgroundColor={muiTheme.palette.colorBrandOrange}
      style={{ boxShadow: 'none', marginRight: 8 }}
      onClick={openDownloadDialog}
      children={<FontIcon className="material-icons">file_download</FontIcon>}
    />
  );
}

DownloadIcon.propTypes = propTypes;

export { DownloadIcon as DownloadIconTest };

export default muiThemeable()(DownloadIcon);
