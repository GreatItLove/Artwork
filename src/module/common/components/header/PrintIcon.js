import React from 'react';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import FontIcon from 'material-ui/FontIcon';
import muiThemeable from 'material-ui/styles/muiThemeable';
import PropTypes from 'prop-types';

const propTypes = {
  muiTheme: PropTypes.shape({
    palette: PropTypes.shape({
      colorBrandOrange: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

/**
 * @function PrintIcon
 * @returns {XML}
 */
function PrintIcon({ muiTheme }) {
  return (
    <FloatingActionButton
      mini
      backgroundColor={muiTheme.palette.colorBrandOrange}
      style={{ boxShadow: 'none', marginRight: 8 }}
      children={<FontIcon className="material-icons">print</FontIcon>}
    />
  );
}

PrintIcon.propTypes = propTypes;

export { PrintIcon as PrintIconTest };

export default muiThemeable()(PrintIcon);
