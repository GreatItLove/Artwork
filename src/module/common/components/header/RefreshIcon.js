import React from 'react';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import FontIcon from 'material-ui/FontIcon';
import PropTypes from 'prop-types';

const propTypes = {
  refresh: PropTypes.func.isRequired,
};

/**
 * @function RefreshIcon
 * @returns {XML}
 */
function RefreshIcon({ refresh }) {
  return (
    <FloatingActionButton
      mini
      backgroundColor='#ffffff'
      onClick={refresh}
      className="panel-refresh"
      children={
        <FontIcon style={{ color: '#1e6291' }} className="material-icons">
          refresh
        </FontIcon>
      }
    />
  );
}

RefreshIcon.propTypes = propTypes;

export default RefreshIcon;
