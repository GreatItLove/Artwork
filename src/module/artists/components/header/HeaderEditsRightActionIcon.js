import React from 'react';

import ActiveToggle from './ActiveToggle';
import TrashIcon from './TrashIcon';

/**
 * @function HeaderEditsRightActionIcon
 * @returns {XML}
 */
function HeaderEditsRightActionIcon() {
  return (
    <div style={{ width: "150px", display: "flex" }}>
      <div style={{ width: "70%", marginTop: "13px", height: "0px" }}>
        <ActiveToggle />
      </div>
      <div style={{ width: "30%" }}>
        <TrashIcon />
      </div>
    </div>
  );
}

export default HeaderEditsRightActionIcon;
