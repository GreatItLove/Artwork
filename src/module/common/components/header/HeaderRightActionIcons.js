import React from 'react';

import DownloadIcon from './DownloadIcon';
import PrintIcon from './PrintIcon';
import AddIcon from './AddIcon';

/**
 * @function HeaderRightActionIcons
 * @returns {XML}
 */
function HeaderRightActionIcons({ openAddDialog, openDownloadDialog }) {
  return (
    <div>
      <DownloadIcon openDownloadDialog={openDownloadDialog} />
      <PrintIcon />
      <AddIcon openAddDialog={openAddDialog} />
    </div>
  );
}

export default HeaderRightActionIcons;
