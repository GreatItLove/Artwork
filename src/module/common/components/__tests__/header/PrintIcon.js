import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import muiTheme from '../../../../../muiTheme';

import { PrintIconTest as PrintIcon } from '../../header/PrintIcon';

describe('PrintIcon', () => {
  it('Snapshot of PrintIcon', () => {
    const wrapper = shallow(<PrintIcon muiTheme={muiTheme} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
