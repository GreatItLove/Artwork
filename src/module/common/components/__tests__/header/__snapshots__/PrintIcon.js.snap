// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintIcon Snapshot of PrintIcon 1`] = `
<FloatingActionButton
  backgroundColor="#e18927"
  disabled={false}
  mini={true}
  secondary={false}
  style={
    Object {
      "boxShadow": "none",
      "marginRight": 8,
    }
  }
  zDepth={2}
>
  <FontIcon
    className="material-icons"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    print
  </FontIcon>
</FloatingActionButton>
`;
